### PR TITLE
test advanced checkpoints

### PR DESCRIPTION
**Pull request by trieloff**
- **test(index): test block visibility, resource loading**
- **test(index): add test for form submissions**

### CODE REVIEW

File: test/it/loadblock.test.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          document.body.appendChild(block);
+
+
+          await new Promise((resolve) => {
+            setTimeout(resolve, 100);
+          });
+
+          await sendMouse({ type: 'click', position: [10, 10] });
+
+          assert(window.called.some((call) => call.checkpoint === 'viewmedia'), 'viewmedia checkpoint missing');
```
Comment (kptdobe) [inline test/it/loadblock.test.html:84]: This is a bad practice: the tests share the same execution context, i.e. they can interfere between each others. This one can for example trigger a `missingresource` you do not check here. If implementation for next test is broken (i.e. fetch does not fire a `missingresouce`, test will be successful but implementation is wrong.

Several todos:
- before / after each test, clean up the `called` stack
- in each test, test for exact equality - if order is random, at least test for window.called.length === 4

If this is not doable in one file, split into multiple files

File: test/it/loadblock.test.html [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            this.skip();
+          }
+
+          const placeholders = await fetch('/test/it/placeholders.json');
+
+          await new Promise((resolve) => {
+            setTimeout(resolve, 100);
+          });
+
+          assert(window.called.some((call) => call.checkpoint === 'missingresource'), 'missingresource checkpoint missing');
```
Comment (kptdobe) [inline test/it/loadblock.test.html:106]: You are testing `missingresource` in a `loadblock` test file. Either rename the file or move to dedicated test file.

File: test/it/loadblock.test.html [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          setTimeout(resolve, 2000);
+        });
+
+        // find the form field
+        form.querySelector('input[type="submit"]').click();
+
+        await new Promise((resolve) => {
+          setTimeout(resolve, 2000);
+        });
+        assert(window.called.some((call) => call.checkpoint === 'formsubmit'), 'formsubmit checkpoint missing');
```
Comment (kptdobe) [inline test/it/loadblock.test.html:123]: Same, testing `formsubmit` in `loadblock`file...

File: test/it/block.plain.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<div>
```
Comment (kptdobe) [inline test/it/block.plain.html:1]: Move file into a `fixtures` subfolder to separate test files and resources (same for image)

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: test/it/block.plain.html (hunk 4 above)
Comment (trieloff) [reply test/it/block.plain.html]: done

File: test/it/loadblock.test.html (hunk 3 above)
Comment (trieloff) [reply test/it/loadblock.test.html]: done

File: test/it/loadblock.test.html (hunk 2 above)
Comment (trieloff) [reply test/it/loadblock.test.html]: done

File: test/it/loadblock.test.html (hunk 1 above)
Comment (trieloff) [reply test/it/loadblock.test.html]: I've split it into multiple files

**Comment by trieloff** [comment]
This brings us up to 98.87% coverage

**Pull request merged**